test/integration/resourcemanager/managedresource/reconciler_test.go

```diff
@@ -0,0 +1,213 @@
+				Expect(err).To(Or(Succeed(), BeAlreadyExistsError()))
+
+				updateTestSecret = &corev1.Secret{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: corev1.SchemeGroupVersion.String(),
+						Kind:       "Secret",
+					}, ObjectMeta: metav1.ObjectMeta{
+						Name:      updateTestSecretName,
+						Namespace: namespaceName,
+					}}
+				Expect(testClient.Get(ctx, client.ObjectKeyFromObject(secretForRef), secretForRef)).To(Or(Succeed(), BeNoMatchError()))
+				secretForRef.Data, err = appendSecretData(updateTestSecret, "update-test-secret.yaml", secretForRef.Data)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testClient.Update(ctx, secretForRef)).To(Succeed())
+
+				Eventually(func() error {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
+					if err != nil {
+						return err
+					}
+					return testClient.Get(ctx, client.ObjectKeyFromObject(updateTestSecret), updateTestSecret)
+				}, time.Minute, 5*time.Second).Should(Succeed())
+
+				Eventually(func() bool {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)
+					condition := gardenerv1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return err == nil && condition != nil && condition.Status == gardencorev1beta1.ConditionTrue
+				}, time.Minute, time.Second).Should(BeTrue())
+			})
+
+			It("the managed resource update should fail", func() {
+				err = testClient.Create(ctx, managedResource)
+				Expect(err).To(Or(Succeed(), BeAlreadyExistsError()))
+
+				updateTestSecret = &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      updateTestSecretName,
+						Namespace: namespaceName,
+					}}
+				Expect(testClient.Get(ctx, client.ObjectKeyFromObject(secretForRef), secretForRef)).To(Or(Succeed(), BeNoMatchError()))
+				secretForRef.Data, err = appendSecretData(updateTestSecret, "update-test-secret.yaml", secretForRef.Data)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testClient.Update(ctx, secretForRef)).To(Succeed())
+
+				Eventually(func() bool {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)
+					condition := gardenerv1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return err == nil && condition != nil && condition.Status == gardencorev1beta1.ConditionFalse
+				}, time.Minute, time.Second).Should(BeTrue())
+			})
+		})
+	})
+})
+
+func createSecretDataFromObject(obj runtime.Object, key string) (*map[string][]byte, error) {
+	jsonObject, err := json.Marshal(obj)
+	if err != nil {
+		return nil, err
+	}
+	data := make(map[string][]byte)
+	data[key] = jsonObject
+	return &data, nil
+}
+
+func appendSecretData(obj runtime.Object, key string, data map[string][]byte) (map[string][]byte, error) {
+	if _, ok := data[key]; ok {
+		return nil, fmt.Errorf("Key: %s already exists", key)
```

Comment:
From https://github.com/golang/go/wiki/CodeReviewComments#error-strings:
> Error strings should not be capitalized [...]

```suggestion
		return nil, fmt.Errorf("key %s already exists", key)
```

Do we need such handling for the test purpose? Do we have `appendSecretData` func call for already existing key?  If not, I would suggest to simply overwrite always the value for the key.